### PR TITLE
fix(provisionPool): update metrics on trades

### LINKED ISSUE #6396: provisionPool metrics are not updated on swap
While working on #6376, @samsiegart noted that after the provisionPool is funded, metrics for `totalMintedConverted` still shows `0n`.

Looks like we don't publish updated metrics after that swap:

https://github.com/Agoric/agoric-sdk/blob/f9b201c03f0a02c76280b70e207324b98561e4b3/packages/vats/src/provisionPool.js#L143-L152

### PR DESCRIPTION
**Pull request by dckc**
closes: #6396

### CODE REVIEW

File: packages/vats/test/test-provisionPool.js [hunk 1]
```diff
@@ -197,7 +198,11 @@ test('provisionPool trades provided assets for IST', async t => {
     initialPoserInvitation,
   );
 
+  // mock gov terms
+  /** @type {Instance} */
+  const electionManager = /** @type {any} */ (null);
   const govTerms = {
+    electionManager,
```
Comment (turadg) [inline packages/vats/test/test-provisionPool.js]:
```suggestion
    electionManager: /** @type {any} */ (null),
```

File: packages/vats/src/provisionPool.js [hunk 2]
```diff
@@ -120,24 +115,47 @@ export const start = async (zcf, privateArgs) => {
...
-    privateArgs.marshaller,
-  );
-  const updateMetrics = walletsProvisioned => {
-    metricsPublisher.publish(
-      harden({
-        walletsProvisioned,
-        totalMintedProvided,
-        totalMintedConverted,
-      }),
+  const makeMetrics = () => {
```
Comment (turadg) [inline packages/vats/src/provisionPool.js:118]: neat encapsulation. We may want to move it back into main scope for durability but we can evaluate that then.

File: packages/vats/src/provisionPool.js [hunk 3]
```diff
@@ -120,24 +115,47 @@ export const start = async (zcf, privateArgs) => {
...
+    const { metricsPublisher, metricsSubscriber } = makeMetricsPublishKit(
+      privateArgs.storageNode,
+      privateArgs.marshaller,
     );
+
+    let walletsProvisioned = 0n;
+    let totalMintedProvided = AmountMath.makeEmpty(poolBrand);
+    let totalMintedConverted = AmountMath.makeEmpty(poolBrand);
+
+    const updateMetrics = () => {
```
Comment (turadg) [inline packages/vats/src/provisionPool.js]: all these onFoo handlers are updating the (local) metrics. let's use "update" to mean revisions to the truth and "publish" to notify subscribers
```suggestion
    const publishMetrics = () => {
```

**Review by turadg — APPROVED** [review]

**Comment by turadg** [comment]
The fix looks good. Some code quality suggestions but more important is to get this into the next rc.

**Comment by samsiegart** [comment]
Verified I can no longer repro the original issue on this patch

**Review by samsiegart — APPROVED** [review]

**Pull request merged**